tests/modules/elmo_test.py

```diff
@@ -131,6 +133,18 @@ def test_elmo(self):
         assert list(elmo_representations[1].size()) == [2, 7, 32]
         assert list(mask.size()) == [2, 7]
 
+    def test_elmo_4D_input(self):
+        # Add an extra dimension to character_ids
+        character_ids = self.character_ids.unsqueeze(1)
+        print(character_ids.size())
+
+        output = self.elmo(character_ids)
+        elmo_representations = output['elmo_representations']
+        mask = output['mask']
+
+        assert list(elmo_representations[0].size()) == [2, 1, 7, 32]
+        assert list(elmo_representations[1].size()) == [2, 1, 7, 32]
+        assert list(mask.size()) == [2, 1, 7]
```

Comment:
For the mask, you can actually check that the elements are what you expect, rather than the shape. Obviously this is harder for the tensors, so no worries about that.